new_webapp/src/components/TopBar.js

```diff
@@ -13,4 +15,24 @@ export const TopBar = () => (
 
 const flavorText = `
 Hoogle+ is a type-driven synthesis engine for Haskell - like Hoogle but able to find compositions of functions. Given a Haskell type, Hoogle+ generates terms that inhabit this type by composing library components. It supports polymorphism, type classes, and higher-order functions. Available library components are listed in the side bar.
-`;
\ No newline at end of file
+`;
+
+const dlStyle = {position:"absolute",
+   top:0,
+   right:0
+}
+
+function dlClick(e) {
+	var element = document.createElement('a');
+	var msgs = log.exportToArray();
+	var text = msgs.join('\n');
```

Comment:
Current JS style would use `const` for things you aren't going to change and `let` for variables you intend to mutate.